### ISSUE TITLE
{macOS} Crash when clicking on the chat room tray notify 

### ISSUE THREAD

**Issue by defnax** [RetroShare/RetroShare#2818]
Crash when clicking on the chat room tray notify 

![image](https://github.com/RetroShare/RetroShare/assets/9952056/2deae8e5-106b-43fb-93cf-bc94e0ed4a28)

![image](https://github.com/RetroShare/RetroShare/assets/9952056/33eb5f0d-8778-4548-bcb1-06c6a2f055cd)

```
1  QCocoaMenu::nsMenu() const                                                  qcocoamenu.mm                                                          102  0x1115c4214    
2  ::-[QNSStatusItem triggerSelector:button:](id, Qt::MouseButton)             qcocoasystemtrayicon.mm                                                415  0x1115d82cc    
3  -[NSWindow(NSEventRouting) _handleMouseDownEvent:isDelayedEvent:]           (x86_64) /System/Library/Frameworks/AppKit.framework/Versions/C/AppKit      0x7fff41280d6d 
4  -[NSWindow(NSEventRouting) _reallySendEvent:isDelayedEvent:]                (x86_64) /System/Library/Frameworks/AppKit.framework/Versions/C/AppKit      0x7fff4127d9c4 
5  -[NSWindow(NSEventRouting) sendEvent:]                                      (x86_64) /System/Library/Frameworks/AppKit.framework/Versions/C/AppKit      0x7fff4127cc70 
6  -[NSStatusBarWindow sendEvent:]                                             (x86_64) /System/Library/Frameworks/AppKit.framework/Versions/C/AppKit      0x7fff410cf885 
7  -[NSApplication(NSEvent) sendEvent:]                                        (x86_64) /System/Library/Frameworks/AppKit.framework/Versions/C/AppKit      0x7fff410de236 
8  ::-[QNSApplication sendEvent:](NSEvent *)                                   qcocoaapplication.mm                                                   174  0x1115c2eba    
9  -[NSApplication run]                                                        (x86_64) /System/Library/Frameworks/AppKit.framework/Versions/C/AppKit      0x7fff4093e87d 
10 QCocoaEventDispatcher::processEvents(QFlags<QEventLoop::ProcessEventsFlag>) qcocoaeventdispatcher.mm                                               429  0x1115bf623    
11 QEventLoop::processEvents(QFlags<QEventLoop::ProcessEventsFlag>)            qeventloop.cpp                                                         138  0x10e148b6f    
12 QEventLoop::exec(QFlags<QEventLoop::ProcessEventsFlag>)                     qeventloop.cpp                                                         225  0x10e148b4e    
13 QCoreApplication::exec()                                                    qcoreapplication.cpp                                                   1400 0x10e14d0f2    
14 main                                                                        main.cpp                                                               590  0x10aeb8f16    
15 start      
```                                                                 (x86_64) /usr/lib/system/libdyld.dylib                                      0x7fff6a627015 



```
Crash log from 2021
```
1 QCocoaMenu::nsMenu() const qcocoamenu.mm 102 0x10bdc4214
2 ::-[QNSStatusItem triggerSelector:button:](id, Qt::MouseButton) qcocoasystemtrayicon.mm 415 0x10bdd82cc
3 -[NSWindow(NSEventRouting) _handleMouseDownEvent:isDelayedEvent:] (x86_64) /System/Library/Frameworks/AppKit.framework/Versions/C/AppKit 0x7fff54febd6d
4 -[NSWindow(NSEventRouting) _reallySendEvent:isDelayedEvent:] (x86_64) /System/Library/Frameworks/AppKit.framework/Versions/C/AppKit 0x7fff54fe89c4
5 -[NSWindow(NSEventRouting) sendEvent:] (x86_64) /System/Library/Frameworks/AppKit.framework/Versions/C/AppKit 0x7fff54fe7c70
6 -[NSStatusBarWindow sendEvent:] (x86_64) /System/Library/Frameworks/AppKit.framework/Versions/C/AppKit 0x7fff54e3a885
7 -[NSApplication(NSEvent) sendEvent:] (x86_64) /System/Library/Frameworks/AppKit.framework/Versions/C/AppKit 0x7fff54e49236
8 ::-[QNSApplication sendEvent:](NSEvent *) qcocoaapplication.mm 174 0x10bdc2eba
9 -[NSApplication run] (x86_64) /System/Library/Frameworks/AppKit.framework/Versions/C/AppKit 0x7fff546a98b5
10 QCocoaEventDispatcher::processEvents(QFlagsQEventLoop::ProcessEventsFlag) qcocoaeventdispatcher.mm 429 0x10bdbf623
11 QEventLoop::processEvents(QFlagsQEventLoop::ProcessEventsFlag) qeventloop.cpp 138 0x108b2db6f
12 QEventLoop::exec(QFlagsQEventLoop::ProcessEventsFlag) qeventloop.cpp 225 0x108b2db4e
13 QMenu::exec(QPoint const&, QAction *) qmenu.cpp 2643 0x107b6452b
14 ChatLobbyUserNotify::iconClicked() ChatLobbyUserNotify.cpp 207 0x105d716e6
15 void doActivate(QObject *, int, void * *) qobject.cpp 3882 0x108b64cc5
16 QSystemTrayIcon::activated(QSystemTrayIcon::ActivationReason) moc_qsystemtrayicon.cpp 241 0x107d66942
17 QSystemTrayIcon::qt_static_metacall(QObject *, QMetaObject::Call, int, void * *) moc_qsystemtrayicon.cpp 124 0x107d66920
18 void doActivate(QObject *, int, void * *) qobject.cpp 3882 0x108b64cc5
19 QPlatformSystemTrayIcon::activated(QPlatformSystemTrayIcon::ActivationReason) moc_qplatformsystemtrayicon.cpp 191 0x10816863b
20 -[NSWindow(NSEventRouting) _handleMouseDownEvent:isDelayedEvent:] (x86_64) /System/Library/Frameworks/AppKit.framework/Versions/C/AppKit 0x7fff54febd6d
```

**Comment by defnax** [comment]
We need some mac users to can test latest macos version, and report the issues, bugs & crashes
some collected issues -> https://github.com/RetroShare/RetroShare/issues/2695

https://retroshare.cc/downloads.html#mac

@magicvince @core-code @cinealfa @rex4539 @Coool @veryprivate @guglez

**Comment by defnax** [comment]
@csoler how can fix the tray menu chat lobby notify? or @thunder2 
it comes from the tray click, it looks like this, but for other tray notifys example for channels,boards, forums does not crash rs.

`14 ChatLobbyUserNotify::iconClicked() ChatLobbyUserNotify.cpp 207 0x105d716e6
`
https://github.com/RetroShare/RetroShare/blob/0caf8e79dbc941fd16eaa15d6d8a20885f5f005e/retroshare-gui/src/gui/chat/ChatLobbyUserNotify.cpp#L207

or i need else disable chat lobby tray notify for macos platfrom to avoid crashes

```
QString strName=tr("Remove All");
QAction *pAction = new QAction( QIcon(), strName, trayMenu);
ActionTag actionTag={0x0, "", true};
pAction->setData(qVariantFromValue(actionTag));
connect(trayMenu, SIGNAL(triggered(QAction*)), this, SLOT(subMenuClicked(QAction*)));
connect(trayMenu, SIGNAL(hovered(QAction*)), this, SLOT(subMenuHovered(QAction*)));
trayMenu->addAction(pAction);

trayMenu->exec(QCursor::pos());
if (doUpdate) updateIcon();`
```

**Comment by thunder2** [comment]
When does it crash? When you click on the tray icon, when you click on a menu item to activate the chat or on "Remove All" ?

**Comment by defnax** [comment]
when i click on menu item , both activate the chat & "remove all"
click on tray icon does not crash

**Comment by thunder2** [comment]
Please try PR 2819

**Comment by defnax** [comment]
tested it but it crashes too

```
1  QCocoaMenu::nsMenu() const                                                  qcocoamenu.mm                                                          102  0x110a44214    
2  ::-[QNSStatusItem triggerSelector:button:](id, Qt::MouseButton)             qcocoasystemtrayicon.mm                                                415  0x110a582cc    
3  -[NSWindow(NSEventRouting) _handleMouseDownEvent:isDelayedEvent:]           (x86_64) /System/Library/Frameworks/AppKit.framework/Versions/C/AppKit      0x7fff44150d6d 
4  -[NSWindow(NSEventRouting) _reallySendEvent:isDelayedEvent:]                (x86_64) /System/Library/Frameworks/AppKit.framework/Versions/C/AppKit      0x7fff4414d9c4 
5  -[NSWindow(NSEventRouting) sendEvent:]                                      (x86_64) /System/Library/Frameworks/AppKit.framework/Versions/C/AppKit      0x7fff4414cc70 
6  -[NSStatusBarWindow sendEvent:]                                             (x86_64) /System/Library/Frameworks/AppKit.framework/Versions/C/AppKit      0x7fff43f9f885 
7  -[NSApplication(NSEvent) sendEvent:]                                        (x86_64) /System/Library/Frameworks/AppKit.framework/Versions/C/AppKit      0x7fff43fae236 
8  ::-[QNSApplication sendEvent:](NSEvent *)                                   qcocoaapplication.mm                                                   174  0x110a42eba    
9  -[NSApplication run]                                                        (x86_64) /System/Library/Frameworks/AppKit.framework/Versions/C/AppKit      0x7fff4380e87d 
10 QCocoaEventDispatcher::processEvents(QFlags<QEventLoop::ProcessEventsFlag>) qcocoaeventdispatcher.mm                                               429  0x110a3f623    
11 QEventLoop::processEvents(QFlags<QEventLoop::ProcessEventsFlag>)            qeventloop.cpp                                                         138  0x10d5cdb6f    
12 QEventLoop::exec(QFlags<QEventLoop::ProcessEventsFlag>)                     qeventloop.cpp                                                         225  0x10d5cdb4e    
13 QCoreApplication::exec()                                                    qcoreapplication.cpp                                                   1400 0x10d5d20f2    
14 main                                                                        main.cpp                                                               590  0x10a33c006    
15 start                                                                       (x86_64) /usr/lib/system/libdyld.dylib                                      0x7fff6d4f7015 

```
![image](https://github.com/RetroShare/RetroShare/assets/9952056/6090e9b2-c80b-4ba4-863f-77f01f05ea2d)

**Comment by defnax** [comment]
alternative for mac we can add separate method
instead with qmenu which lets crash rs, like how for forums/channels

open right chat widget with the right chat room, dont know how to navigate to the right chat room? @thunder2 
```
#ifdef Q_OS_MACOS
void ChatLobbyUserNotify::iconClickedonMac(QAction* action)
{
   MainWindow::showWindow(MainWindow::ChatLobby);
}
#endif // Q_OS_MACOS
```
```
void ChatLobbyUserNotify::iconClicked()
{
        #ifdef Q_OS_MACOS
        MainWindow::showWindow(MainWindow::ChatLobby);
        #else

	/// Tray icon Menu ///
	QMenu* trayMenu = new QMenu(MainWindow::getInstance());
	std::list<ChatLobbyId> lobbies;
	rsMsgs->getChatLobbyList(lobbies);
	bool doUpdate=false;

        .....
       #endif
}
```

**Comment by defnax** [comment]
- its now fixed #2820
- alternative fix
